### PR TITLE
Fix compresskv with lookahead issue

### PR DESCRIPTION
**Pull request by cyita**
## Description

Fix attn_mask in compresskv w/ lookahead.
Note: chatglm is partially fixed and will print abnormal output.

### 4. How to test?
- [ ] N/A
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/models/qwen2.py [hunk 1]
```diff
@@ -472,8 +472,9 @@ def qwen2_attention_forward(
                            is_causal=True).to(hidden_states.dtype)
     elif use_sdp(q_len, kv_seq_len, self.head_dim, query_states):
         import xe_addons
-        if use_compresskv:
-            attention_mask = None
+        if use_compresskv and attention_mask is not None:
+            context_len = key_states.size(2)
+            attention_mask = attention_mask[:, :, :, -context_len:]
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/models/qwen2.py:477]: Make it a method of `DynamicCompressCache` or a utility method

**Review by rnwang04 — APPROVED** [review]

**Comment by cyita** [comment]
PR Validation: https://github.com/intel-analytics/ipex-llm-workflow/actions/runs/10349724071

File: python/llm/src/ipex_llm/transformers/models/chatglm2.py [hunk 2]
```diff
@@ -300,6 +303,8 @@ def chatglm2_attention_forward(
     attn_weights = None
     if use_sdp(q_len, kv_seq_len, head_dim, query_states):
         import xe_addons
+        if use_compresskv and attention_mask is not None:
+            attention_mask = None
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/models/chatglm2.py:307]: Do you want to set it to None here?

Comment (cyita) [reply python/llm/src/ipex_llm/transformers/models/chatglm2.py:307]: Yes, the attn_mask of chatglm2 is different from other models in lookahead, and needs to be fixed in other pr.

**Pull request merged**